.github/build.in.yml

```
@@ -12,19 +12,14 @@ env:
   # not necessarily satisfy this property.
   LAKE_NO_CACHE: true
 
-jobs:
-  # Cancels previous runs of jobs in this file
-  cancel:
-    if: github.repository == 'leanprover-community/mathlib4'
-    name: 'Cancel Previous Runs (CI)'
-    runs-on: ubuntu-latest
-    # timeout-minutes: 3
-    steps:
-      - uses: styfle/cancel-workflow-action@0.12.1
-        with:
-          all_but_latest: true
-          access_token: ${{ github.token }}
+concurrency:
+  # label each workflow run; only the latest with each label will run
+  # workflows on master get more expressive labels
+  group: ${{ github.workflow }}-${{ github.ref }}.${{(github.ref == 'refs/heads/master' && github.run_id) || ''}}
+  # cancel any running workflow with the same label
+  cancel-in-progress: true
```

After testing, I'm pretty sure that the `concurrency` feature does what we want it to.

I think what you're doing is equivalent, but [GitHub's docs use something like this](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-only-cancel-in-progress-jobs-on-specific-branches) to avoid `cancel-in-progress` on a certain branch:
```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
  # cancel any workflows with the same label not running on master
  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```
To me it looks somewhat easier to understand, but I don't mind if this is merged as is.

Thanks again for getting this fixed!
bors d+

---

the reason i did this different from the example is because *in a concurrency group, only one workflow runs at a time*. That means that with the suggested version, it would not build multiple versions of master at the same time, since they're in the same concurrency group.

bors r+